### PR TITLE
Microsoft.AAD/DomainServices Swagger: Add new api-version 2017-06-01

### PR DESCRIPTION
**Pull request by jihochang**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jihochang** [comment]
Previous Pull Request accepted by veronicagg

https://github.com/Azure/azure-rest-api-specs/pull/1689#issuecomment-330640661

**Comment by sergey-shandar** [comment]
@ravbhatnagar new API version `DomainServicesSwagger.json`

**Comment by jihochang** [comment]
The api-version itself is not new. We are just adding the swagger for it now.

File: specification/domainservices/resource-manager/readme.md [hunk 1]
```diff
@@ -39,6 +39,14 @@ input-file:
 - Microsoft.AAD/2017-01-01/domainservices.json
 ```
 
+### Tag: package-2017-06
+
+These settings apply only when `--tag=package-2017-06` is specified on the command line.
+
+``` yaml $(tag) == 'package-2017-06'
+input-file:
+- Microsoft.AAD/2017-06-01/domainservices.json
```
Comment (olydis) [inline specification/domainservices/resource-manager/readme.md]: note that this path is *wrong*, the filename is `DomainServicesSwagger.json`. I've just fixed that for the previous API version, please make sure you also adjust this accordingly.

**Comment by jihochang** [comment]
I renamed the swagger file to match what is in the readme file.

**Comment by jihochang** [comment]
Could you take another look and approve the pull request? Our ARM swagger deadline for this is tomorrow. This is an already existing API-VERSION and we are just now adding the swagger files corresponding to it.

**Comment by jihochang** [comment]
Ping any update on this?

**Comment by ravbhatnagar** [comment]
Looks good.

File: specification/domainservices/resource-manager/Microsoft.AAD/2017-01-01/domainservices.json [hunk 2]
```diff
@@ -26,24 +26,30 @@
...
+		"produces": [
+          "application/json"
         ],
         "responses": {
           "200": {
-            "description": "HTTP 200 (OK) if the operation was successful.",
+            "description": "The operation succeeded. The response contains the possible operations on the Microsoft.AAD resource provider.",
             "schema": {
-              "$ref": "#/definitions/OperationEntityListResult"
+              "$ref": "#/definitions/OperationListResult"
```
Comment (sergey-shandar) [inline specification/domainservices/resource-manager/Microsoft.AAD/2017-01-01/domainservices.json]: This is definitely an SDK breaking changes. Is there a significant reason to do these breaking changes in the old version of API?

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

**Comment by jihochang** [comment]
Creating new pull request, due to bad merge. 

New Pull Request: https://github.com/Azure/azure-rest-api-specs/pull/1790

**Pull request closed without merging**